Bindings/Java/Matlab/Utilities/osimC3D.m

```diff
@@ -81,11 +120,86 @@ function rotateData(obj,axis,value)
             % rotate the tables
             obj.rotateTable(obj.markers, axis, value);
             obj.rotateTable(obj.forces, axis, value);
-            
             disp('Marker and Force tables have been rotated')
         end
+        function writeTRC(obj,varargin)
+            % Write marker data to trc file.
+            % osimC3d.writeTRC()                       Write to current dir using c3d name
+            % osimC3d.writeTRC('Walking.trc')          Write to current dir with input name
+            % osimC3d.writeTRC('C:\data\Walking.trc')  Write to input location from full input path
+
+            % Compute an output path to use for writing to file
+            outputPath = generateOutputPath(obj,varargin,'.trc');
+
+
+            import org.opensim.modeling.*
+            % Write to file
+            TRCFileAdapter().write( obj.getTable_markers, outputPath)
+            disp(['Marker file written to ' outputPath]);
+        end
+        function writeMOT(obj,varargin)
+         % Write forces data to mot file
+         % osimC3d.writeMOT()                       Write to current dir using c3d name
+         % osimC3d.writeMOT('Walking.mot')          Write to current dir with input name
+         % osimC3d.writeMOT('C:\data\Walking.mot')  Write to input location from full input path
+
+         % Compute an output path to use for writing to file
+         outputPath = generateOutputPath(obj,varargin,'.mot');
+         
+         import org.opensim.modeling.*
+         % Get the forces table
+         forces = obj.getTable_forces();
+         % Get the column labels
+         labels = forces.getColumnLabels();
+         % make a copy
+         updlabels = labels; 
+          
+         for i = 0 : labels.size() - 1
+            % Get the label as a string
+            label = char(labels.get(i));
+            % Transform the label depending on force, point, or moment
+            if ~isempty(strfind(label,'f'))
+                label = strrep(label,'f', 'ground_force_');
+                label = [label '_v'];
+            elseif ~isempty(strfind(label,'p'))
+                label = strrep(label,'p', 'ground_force_');
+                label = [label '_p'];
+            elseif ~isempty(strfind(label,'m'))
+                label = strrep(label,'m', 'ground_moment_');
+                label = [label '_m'];
+            end
```

Comment:
I'm having trouble following this part. What do the labels look like beforehand? Do they look something like `f1`, `f2`, `f3`?

Reply:
I added a comment, let me know if it makes sense or not :/

Reply:
Yes that makes sense.